javascript/javascript1/week4/VoiceAssistant.js

```diff
@@ -0,0 +1,74 @@
+const myObj = {
+    name: 0,
+    toDo: []
+}
+function getReply(command) {
+    if (command.split(' ').shift() === 'Set') {
+        let time = command.split('Set a timer for ').pop().split(' minutes')[0];
+        console.log(`Timer set for ${time} minutes`);
+        setTimeout(message, (time*60)*1000);
+        function message() {
+            console.log('Timer done');
+        }
+    }
+    else if (command.split(' ').shift() === 'Hello') {
+        myObj.name = command.split(' ').pop();
+        console.log(`Nice to meet you ${myObj.name}`)
+    }
+    else if (command === 'What is my name' && myObj.name === 0) {
+        console.log("Please, define your name with command getReply('Hello my name is...') ")
+
+    }
+    else if (command === 'What is my name' && myObj.name !== 0) {
+        console.log(`Your name is ${myObj.name}`)
+    }
+    else if (command.split(' ').shift() === 'Add') {
+        const inputStr = command;
+        myObj.toDo.push(inputStr.split('Add ').pop().split(' to')[0]);
+        console.log(`${inputStr.split('Add ').pop().split(' to')[0]} added to your todo`)
+    }
+    else if (command.split(' ').shift() === 'Remove') {
+        const inputStr = command;
+        toDoRemove = inputStr.split('Remove ').pop().split(' from')[0];
+        const index = myObj.toDo.indexOf(toDoRemove);
+        if (index > -1) {
+            myObj.toDo.splice(index, 1);
+            console.log(`Removed ${toDoRemove} from your todo`);
+        }
+        else {
+            console.log(`${toDoRemove} was not found in your todo`)
+        }
+    }
+    else if (command === 'What day is it today?') {
+        let d = new Date();
+        let ye = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(d);
+        let mo = new Intl.DateTimeFormat('en', { month: 'long' }).format(d);
+        let da = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(d);
+        console.log(`${da}. Of ${mo} ${ye}`);
+    }
+    else if (typeof eval(command.split(' ').pop()) === "number") {
```

Comment:
Hope you've read that eval is evil and we should never-never-never-ever use it :) 
Just for fun you can try
```javascript
getReply('What is alert("Hello from eval")')
```